src/webview/html/messageAsHtml.js

```diff
@@ -72,13 +78,90 @@ $!${messageReactionListAsHtml(reactions, ownUser.user_id, allImageEmojiById)}
 `;
 };
 
-const widgetBody = (message: Message | Outbox) => template`
+/**
+ * Render the body of a message that has submessages.
+ *
+ * Must not be called on a message without any submessages.
+ */
+const widgetBody = (message: Message, ownUserId: UserId) => {
+  invariant(
+    message.submessages !== undefined && message.submessages.length > 0,
+    'should have submessages',
+  );
+
+  const sortedSubmessages = message.submessages.concat();
+  sortedSubmessages.sort((m1, m2) => m1.id - m2.id);
```

Comment:
Ah neat, I hadn't seen `Array#concat` used for this.

Another thing that can simplify this is that `Array#sort` returns the array. So there's no need to initialize first and then sort; can just initialize with the result of `sort`.

Then in turn that expression can just be inlined in the one place we're using this.